101-vm-simple-rhel/azuredeploy.json

```diff
@@ -24,25 +24,31 @@
       "metadata": {
         "description": "The RHEL version for the VM. This will pick the latest available image of this given RHEL version."
       }
+    },
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name for the Virtual Machine."
+      }
     }
   },
   "variables": {
-    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'rhelvm')]",
-    "dataDisk1VhdName": "datadisk1",
+    "storageAccountName": "[concat(uniquestring(parameters('vmName')), 'storage')]",
+    "dataDisk1VhdName": "[concat(uniquestring(parameters('vmName')), 'datadisk1')]",
+    "dataDisk2VhdName": "[concat(uniquestring(parameters('vmName')), 'datadisk2')]",
     "imagePublisher": "RedHat",
     "imageOffer": "RHEL",
-    "OSDiskName": "osdiskforrhelsimple",
-    "nicName": "myVMNic",
+    "OSDiskName": "[concat(uniquestring(parameters('vmName')), 'osdisk')]",
+    "nicName": "[concat(uniquestring(parameters('vmName')), 'nic')]",
     "addressPrefix": "10.0.0.0/16",
     "subnetName": "Subnet",
     "subnetPrefix": "10.0.0.0/24",
     "storageAccountType": "Standard_LRS",
-    "publicIPAddressName": "myPublicIP",
+    "publicIPAddressName": "[concat(uniquestring(parameters('vmName')), 'publicip')]",
     "publicIPAddressType": "Dynamic",
     "vmStorageAccountContainerName": "vhds",
-    "vmName": "MyRHELVM",
-    "vmSize": "Standard_D1",
-    "virtualNetworkName": "MyVNET",
+    "vmSize": "Standard_DS2",
+    "virtualNetworkName": "[concat(uniquestring(parameters('vmName')), 'vnet')]",
     "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
     "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
     "apiVersion": "2015-06-15"
```

Comment:
Having variables for "apiVersion" values breaks the JSON schema validation, which means users don't get auto-completion and error checking while they're editing their deployment template. Please don't save "apiVersion" values as parameters or variables.